source/material_model/melt_boukare.cc

```diff
@@ -0,0 +1,1260 @@
+                                                     * this->get_gravity_model().gravity_vector(in.position[q]);
+            }
+
+
+          if (this->include_melt_transport())
+            {
+              // Calculate the melting rate
+              const unsigned int Fe_solid_idx = this->introspection().compositional_index_for_name("molar_Fe_in_solid");
+              const unsigned int Fe_melt_idx = this->introspection().compositional_index_for_name("molar_Fe_in_melt");
+              const unsigned int porosity_idx = this->introspection().compositional_index_for_name("porosity");
+
+              // TODO: Alternatively, this could also be done by summing over endmember_mole_fractions_in_composite
+              const double old_porosity = in.composition[q][porosity_idx];
+              const double old_solid_composition = in.composition[q][Fe_solid_idx];
+              const double old_melt_composition = in.composition[q][Fe_melt_idx];
+              const double old_melt_molar_fraction = melt_molar_fraction;
+
+              // in this simple model, the bulk composition is just one number, namely
+              // the molar fraction of the combined iron endmembers
+              const double bulk_composition = old_melt_composition * melt_molar_fraction + old_solid_composition * solid_molar_fraction;
+              double molar_volatiles_in_melt = 0.0;
+              double solid_composition, melt_composition;
+
+              // calculate the melting rate as difference between the equilibrium melt fraction
+              // and the solution of the previous time step, and also update melt and solid composition
+              melt_molar_fraction = melt_fraction(in.temperature[q],
+                                                  this->get_adiabatic_conditions().pressure(in.position[q]),
+                                                  bulk_composition,
+                                                  molar_volatiles_in_melt,
+                                                  solid_composition,
+                                                  melt_composition);
+
+              if (boukare_out != nullptr)
+                {
+                  boukare_out->bulk_composition[q] = bulk_composition;
+                  boukare_out->molar_volatiles_in_melt[q] = molar_volatiles_in_melt;
+                }
+
+              // We have to compute the update to the melt fraction in such a way that the bulk composition is conserved.
+              const double change_of_melt_composition = reaction_fraction * limit_update_to_0_and_1(old_melt_composition, melt_composition - old_melt_composition);
+              const double change_of_melt_fraction = reaction_fraction * limit_update_to_0_and_1(old_melt_molar_fraction, melt_molar_fraction - old_melt_molar_fraction);
+
+              melt_molar_fraction = old_melt_molar_fraction + change_of_melt_fraction;
+              melt_composition = old_melt_composition + change_of_melt_composition;
```

Comment:
reverse the order of these two lines to be consistent with the two lines directly above